### ISSUE TITLE
Pydantic conversions are not supported

### ISSUE THREAD

**Issue by vberlier** [microsoft/pyright#2830]
A recent Pyright version introduced type-checking for arguments provided to the constructor of Pydantic models. Unfortunately, Pydantic is supposed to also convert any value into the desired shape if possible, and this means that nested models can be expressed as dictionaries.

```python
class Bar(BaseModel):
    thing: str

class Foo(BaseModel):
    bar: Bar

# Argument of type "dict[str, str]" cannot be assigned to parameter "bar" of type "Bar" in function 
Foo(bar={"thing": "hello"})
```

This is a valid usage an therefore shouldn't cause any error.

**Comment by erictraut** [comment]
Yes, this is a known limitation of the dataclass_transform mechanism that we introduced. We don't currently have any plans to support features that are specific to pydantic. If you want to use pyright as a type checker, I recommend that you avoid using this feature of pydantic.
